cli/internal/kubecmd/kubecmd.go

```diff
@@ -193,35 +194,35 @@ func (k *KubeCmd) GetClusterAttestationConfig(ctx context.Context, variant varia
 	return existingAttestationConfig, nil
 }
 
-// UpdateAttestationConfig updates the Constellation cluster's attestation config.
-// A backup of the previous config is created before updating.
-func (k *KubeCmd) UpdateAttestationConfig(ctx context.Context, newAttestConfig config.AttestationCfg) error {
-	// backup of previous measurements
-	joinConfig, err := k.kubectl.GetConfigMap(ctx, constants.ConstellationNamespace, constants.JoinConfigMap)
+// ApplyAttestationConfig creates or updates the Constellation cluster's attestation config.
+func (k *KubeCmd) ApplyAttestationConfig(ctx context.Context, newAttestConfig config.AttestationCfg, measurementSalt []byte) error {
+	newConfigJSON, err := json.Marshal(newAttestConfig)
 	if err != nil {
-		return fmt.Errorf("getting %s ConfigMap: %w", constants.JoinConfigMap, err)
-	}
-
-	// create backup of previous config
-	backup := joinConfig.DeepCopy()
-	backup.ObjectMeta = metav1.ObjectMeta{}
-	backup.Name = fmt.Sprintf("%s-backup", constants.JoinConfigMap)
-	if err := k.applyConfigMap(ctx, backup); err != nil {
-		return fmt.Errorf("creating backup of join-config ConfigMap: %w", err)
+		return fmt.Errorf("marshaling attestation config: %w", err)
 	}
-	k.log.Debugf("Created backup of %s ConfigMap %q in namespace %q", constants.JoinConfigMap, backup.Name, backup.Namespace)
 
-	newConfigJSON, err := json.Marshal(newAttestConfig)
+	joinConfig, err := k.kubectl.GetConfigMap(ctx, constants.ConstellationNamespace, constants.JoinConfigMap)
 	if err != nil {
-		return fmt.Errorf("marshaling attestation config: %w", err)
+		if !k8serrors.IsNotFound(err) {
+			return fmt.Errorf("getting %s ConfigMap: %w", constants.JoinConfigMap, err)
+		}
+
+		k.log.Debugf("ConfigMap %q does not exist in namespace %q, creating it now", constants.JoinConfigMap, constants.ConstellationNamespace)
+		if err := k.kubectl.CreateConfigMap(ctx, joinConfigMap(newConfigJSON, measurementSalt)); err != nil {
```

Comment:
why not use Apply instead of Create + Update?

Reply:
Because we are doing two separate things here.
If a configmap exists, we simply want to update 2 keys in that configmap and update it.
If it does not, we create a new one.
The logic here does not fit well with calling `CoreV1().ConfigMaps().Apply()`

Reply:
Why shouldn't this work?

cm, errNotExist := Get()
if errNotExist {
 cm = joinConfigMap(newConfigJSON, measurementSalt)
}
cm.Data =
cm.BinaryData = 
Apply(cm)